pkg/scheduler/uid/uid.go

```diff
@@ -15,55 +15,46 @@ type UID struct {
 	ser   string
 }
 
-func New(group uint64, name string) *UID {
-	if name == "" {
-		name = uuid.New()
-	}
-	return &UID{
+func New(group uint64, name string) UID {
+	return UID{
```

Comment:
why eliminate the check for an empty name? maybe return an error in that case? it was my intent that name would never be empty